types/src/primary.rs

```diff
@@ -433,6 +433,10 @@ impl Certificate {
     pub fn origin(&self) -> PublicKey {
         self.header.author.clone()
     }
+
+    pub fn is_empty(&self) -> bool {
+        self.header.payload.is_empty()
+    }
```

Comment:
Can we add 
1. a scary warning on the definition of `is_empty` saying it is now load-bearing for the liveness of the protocol, and its semantics should check there are >0 transactions in the header?
2. can we add a test for this function?
3. can we open an issue noting that it would be enough to create empty batches, to fool folks into thinking headers are not? we need to length-prefix our hashes anyway, this makes me think we want a `tx_count` in the header to resolve this.

Reply:
See also https://github.com/MystenLabs/narwhal/issues/517

Reply:
I added the ping in the proposer directly, where you thinking to put it in the function you linked?

Reply:
I'm not sure I understand the question?

Reply:
ha, that is because it makes no sense :)

I mean `ping -> print`. I added a tracing/log print in the proposer to warn about empty headers